yt_dlp/extractor/digiview.py

```diff
@@ -0,0 +1,112 @@
+                'channel': 'Philipp Hagemeister',
+                'channel_id': 'UCLqxVugv74EIW3VWh2NOa3Q',
+                'channel_url': r're:https?://(?:www\.)?youtube\.com/channel/UCLqxVugv74EIW3VWh2NOa3Q',
+                'upload_date': '20121002',
+                'description': 'md5:8fb536f4877b8a7455c2ec23794dbc22',
+                'categories': ['Science & Technology'],
+                'tags': ['youtube-dl'],
+                'duration': 10,
+                'view_count': int,
+                'like_count': int,
+                'availability': 'public',
+                'playable_in_embed': True,
+                'thumbnail': 'https://i.ytimg.com/vi/BaW_jenozKc/maxresdefault.jpg',
+                'live_status': 'not_live',
+                'age_limit': 0,
+                'comment_count': int,
+                'channel_follower_count': int,
+                'uploader': 'Philipp Hagemeister',
+                'uploader_url': 'https://www.youtube.com/@PhilippHagemeister',
+                'uploader_id': '@PhilippHagemeister',
+                'heatmap': 'count:100',
+            },
+        },
+        {
+            # cut video
+            'url': 'https://ladigitale.dev/digiview/#/v/663e17f2f3f18',
+            'md5': 'acdf2c99c1e4d67664c9fbc5695986a9',
+            'info_dict': {
+                'id': 'BaW_jenozKc',
+                'ext': 'mp4',
+                'title': 'youtube-dl test video "\'/\\ä↭𝕐',
+                'channel': 'Philipp Hagemeister',
+                'channel_id': 'UCLqxVugv74EIW3VWh2NOa3Q',
+                'channel_url': r're:https?://(?:www\.)?youtube\.com/channel/UCLqxVugv74EIW3VWh2NOa3Q',
+                'upload_date': '20121002',
+                'description': 'md5:8fb536f4877b8a7455c2ec23794dbc22',
+                'categories': ['Science & Technology'],
+                'tags': ['youtube-dl'],
+                'duration': 3,
+                'view_count': int,
+                'like_count': int,
+                'availability': 'public',
+                'playable_in_embed': True,
+                'thumbnail': 'https://i.ytimg.com/vi/BaW_jenozKc/maxresdefault.jpg',
+                'live_status': 'not_live',
+                'age_limit': 0,
+                'comment_count': int,
+                'channel_follower_count': int,
+                'uploader': 'Philipp Hagemeister',
+                'uploader_url': 'https://www.youtube.com/@PhilippHagemeister',
+                'uploader_id': '@PhilippHagemeister',
+                'heatmap': 'count:100',
+            },
+        },
+    ]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage_data = self._download_json(
+            Request(
+                'https://ladigitale.dev/digiview/inc/recuperer_video.php',
+                data=urllib.parse.urlencode({'id': video_id}).encode(),
+                method='POST',
+            ),
+            video_id,
+        )
```

Comment:
I don't understand why you wrap this in a Request object. Can't you do
```suggestion
        video_data = self._download_json(
            'https://ladigitale.dev/digiview/inc/recuperer_video.php', video_id,
            data=urllib.parse.urlencode({'id': video_id}).encode())
```

Reply:
I absolutely need to make a POST request, see https://codeberg.org/ladigitale/digiview/src/commit/b10f7348f7/inc/recuperer_video.php#L42.

Reply:
if you pass any `data` value besides `None` to the `_download_json` helper method it will make it a POST request for you

Reply:
I would use `urlencode_postdata` here though instead of `urllib.parse` (import from `utils`)

Reply:
edited, refresh the page